sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -129,6 +130,34 @@ class RelationalGroupedDataset protected[sql](
     (inputExpr: Expression) => exprToFunc(inputExpr)
   }
 
+  /**
+   * Returns a `KeyValueGroupedDataset` where the data is grouped by the grouping expressions
+   * of current `RelationalGroupedDataset`.
+   *
+   * @since 3.0.0
+   */
+  def as[U: Encoder, T: Encoder]: KeyValueGroupedDataset[U, T] = {
+    val keyEncoder = encoderFor[U]
+    val valueEncoder = encoderFor[T]
+
+    val aliasedGrps = groupingExprs.map { g =>
```

Comment:
can we create a fake project to hold the grouping expressions and run the analyzer to resolve?